### PR TITLE
Zenscrapeinit

### PR DESCRIPTION
**Pull request by abhishek9celestial**
Zenscrape API Test cases for below pages.

1. Pricing
2. Blog
3. Documentation
4. Status

### CODE REVIEW

File: cypress/e2e/zenscrape/documentation.cy.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    it('1. Test to accept cookies if not already accepted', () => {
+        cy.AcceptCookies();
+    });
+
+    it('2. Test to click the documentation hyperlink', () => {
+        cy.get('.gap-8 > [href="//app.zenscrape.com/documentation"]').click();
+    });
+
+    it('3. Test to check the text in the documentation section', () => {
+        cy.wait(8000);
```
Comment (NaveenK15) [inline cypress/e2e/zenscrape/documentation.cy.js]: why wait?

Comment (NaveenK15) [reply cypress/e2e/zenscrape/documentation.cy.js]: can you try cy.document().its('readyState').should('eq', 'complete');

File: cypress/e2e/zenscrape/documentation.cy.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    xit('8. Test to check the "Run in Postman" button functionality', () => {
+        cy.get('.postman-run-button').click();
+        cy.wait(5000);
+        //Added an 5-second wait to allow the postman web page to load. A more reliable solution should be implemented to handle this wait.
+        cy.url().should('eq', 'https://app.getpostman.com/run-collection/85f8b936e3e1870c3fa9');
+        cy.go('back');
+    });
+
+    it('9. Test to navigate to the "Credit Costs & Failed Requests" section', () => {
+        cy.get('.mb-3 > :nth-child(1) > .nav-link').click();
```
Comment (NaveenK15) [inline cypress/e2e/zenscrape/documentation.cy.js:43]: what if the hierarchy changes ?

File: cypress/e2e/zenscrape/register.cy.js [hunk 3]
```diff
@@ -12,11 +12,11 @@ describe('Zenscrape Register Page  ', () => {
         cy.get('.rounded-md > .shadow').click();
     });
 
-    it('2. Test to redirect to the Register page', () => {
+    it('2. Test to check the to the register page', () => {
         cy.navigateUrlwithCookies('https://app.zenscrape.com/register');
     });
 
-    it('3. Test to Verify subscription text is there or not', () => {
+    it('3. Test to Verify subscription text is present or not', () => {
```
Comment (NaveenK15) [inline cypress/e2e/zenscrape/register.cy.js]: or not ?? 
To verify that a "Subscription" text is present

File: cypress/e2e/zenscrape/register.cy.js [hunk 4]
```diff
@@ -28,135 +28,135 @@ describe('Zenscrape Register Page  ', () => {
         cy.get(':nth-child(5) > :nth-child(1) > .d-flex').should('be.visible');
     });
 
-    it('5. Test to Verify subscription monthly or yearly option availble or not', () => {
+    it('5. Test to verify subscription for monthly or yearly option availble or not', () => {
```
Comment (NaveenK15) [inline cypress/e2e/zenscrape/register.cy.js]: or not everywhere.

File: cypress/e2e/zenscrape/register.cy.js [hunk 5]
```diff
@@ -28,135 +28,135 @@ describe('Zenscrape Register Page  ', () => {
...
         cy.wait(500);
     });
 
-    it('14. Test to Verify the "Features" button of Large Monthly plan and should open the popup for Large Monthly plan', () => {
+    it('14. Test to verify the "Features" button of large monthly plan and should open the popup for large monthly plan', () => {
         cy.get(':nth-child(4) > :nth-child(2) > .btn').contains('Features').click({ force: true });
         cy.get('.modal-header').contains('Large');
         cy.get('.plan-feature-list > li').should('be.visible', 'contain.text', '3.000.000 requests per month');
         cy.get('.modal-footer > .btn').click({ force: true });
         cy.wait(500);
```
Comment (NaveenK15) [inline cypress/e2e/zenscrape/register.cy.js]: explain this wait

File: cypress/e2e/zenscrape/register.cy.js [hunk 6]
```diff
@@ -28,135 +28,135 @@ describe('Zenscrape Register Page  ', () => {
...
     });
 
-    it('15. Test to Verify the "Features" button of Professional Monthly plan and should open the popup for Professional Monthly plan', () => {
+    it('15. Test to verify the "Features" button of professional monthly plan and should open the popup for professional monthly plan', () => {
         cy.get(':nth-child(5) > :nth-child(2) > .btn').contains('Features').click({ force: true });
         cy.get('.modal-header').contains('Professional');
         cy.get('.plan-feature-list > li').should('be.visible', 'contain.text', '5 000 000 requests per month');
         cy.wait(500);
```
Comment (NaveenK15) [inline cypress/e2e/zenscrape/register.cy.js]: why wait

File: cypress/e2e/zenscrape/documentation.cy.js (hunk 1 above)
Comment (abhishek9celestial) [reply cypress/e2e/zenscrape/documentation.cy.js]: Removed the cy.wait and it's working fine

File: cypress/e2e/zenscrape/documentation.cy.js (hunk 2 above)
Comment (abhishek9celestial) [reply cypress/e2e/zenscrape/documentation.cy.js:43]: There is no way to get the element other than selector and the selector is having the class like this only.

File: cypress/e2e/zenscrape/register.cy.js (hunk 3 above)
Comment (abhishek9celestial) [reply cypress/e2e/zenscrape/register.cy.js]: removed or not

File: cypress/e2e/zenscrape/register.cy.js (hunk 4 above)
Comment (abhishek9celestial) [reply cypress/e2e/zenscrape/register.cy.js]: removed

File: cypress/e2e/zenscrape/register.cy.js (hunk 5 above)
Comment (abhishek9celestial) [reply cypress/e2e/zenscrape/register.cy.js]: need wait in order to get the modal close button.

File: cypress/e2e/zenscrape/register.cy.js (hunk 6 above)
Comment (abhishek9celestial) [reply cypress/e2e/zenscrape/register.cy.js]: need wait in order to get the modal close button.

File: cypress/e2e/zenscrape/register.cy.js [hunk 7]
```diff
@@ -1,162 +1,167 @@
...
     });
 
-    it('12. Test to Verify the "Features" button of Small Monthly plan and should open the popup for Small Monthly plan', () => {
+    it('12. Test to verify the "Features" button of "Small" monthly plan and should open the popup for "Small" monthly plan', () => {
         cy.get(':nth-child(2) > :nth-child(2) > .btn').contains('Features').click({ force: true })
-        cy.get('.modal-header').contains('Small');
+        cy.get('.modal-header').contains('Small').should('be.visible');
         cy.get('.plan-feature-list > li').should('be.visible', 'contain.text', '250.000 requests per month');
         cy.get('.modal-footer > .btn').click({ force: true });
         cy.wait(500);
```
Comment (NaveenK15) [inline cypress/e2e/zenscrape/register.cy.js]: add ready state event

File: cypress/e2e/zenscrape/register.cy.js [hunk 8]
```diff
@@ -1,162 +1,167 @@
...
     });
 
-    it('13. Test to Verify the "Features" button of Medium Monthly plan and should open the popup for Medium Monthly plan', () => {
+    it('13. Test to verify the "Features" button of "Medium" monthly plan and should open the popup for "Medium" monthly plan', () => {
         cy.get(':nth-child(3) > :nth-child(2) > .btn').contains('Features').click({ force: true });
-        cy.get('.modal-header').contains('Medium');
+        cy.get('.modal-header').contains('Medium').should('be.visible');
         cy.get('.plan-feature-list > li').should('be.visible', 'contain.text', '1.000.000 requests per month');
         cy.get('.modal-footer > .btn').click({ force: true });
         cy.wait(500);
```
Comment (NaveenK15) [inline cypress/e2e/zenscrape/register.cy.js]: ?

File: cypress/e2e/zenscrape/register.cy.js [hunk 9]
```diff
@@ -1,162 +1,167 @@
...
     });
 
-    it('14. Test to Verify the "Features" button of Large Monthly plan and should open the popup for Large Monthly plan', () => {
+    it('14. Test to verify the "Features" button of "Large" monthly plan and should open the popup for "Large" monthly plan', () => {
         cy.get(':nth-child(4) > :nth-child(2) > .btn').contains('Features').click({ force: true });
-        cy.get('.modal-header').contains('Large');
+        cy.get('.modal-header').contains('Large').should('be.visible');
         cy.get('.plan-feature-list > li').should('be.visible', 'contain.text', '3.000.000 requests per month');
         cy.get('.modal-footer > .btn').click({ force: true });
         cy.wait(500);
```
Comment (NaveenK15) [inline cypress/e2e/zenscrape/register.cy.js]: ?

File: cypress/e2e/zenscrape/register.cy.js [hunk 10]
```diff
@@ -1,162 +1,167 @@
...
+        //Todo: Added an 0.5 second wait to load the features pop up. A more reliable solution should be implemented to handle this wait.
     });
 
-    it('15. Test to Verify the "Features" button of Professional Monthly plan and should open the popup for Professional Monthly plan', () => {
+    it('15. Test to verify the "Features" button of "Professional" monthly plan and should open the popup for "Professional" monthly plan', () => {
         cy.get(':nth-child(5) > :nth-child(2) > .btn').contains('Features').click({ force: true });
-        cy.get('.modal-header').contains('Professional');
+        cy.get('.modal-header').contains('Professional').should('be.visible');
         cy.get('.plan-feature-list > li').should('be.visible', 'contain.text', '5 000 000 requests per month');
         cy.wait(500);
```
Comment (NaveenK15) [inline cypress/e2e/zenscrape/register.cy.js]: ?

File: cypress/e2e/zenscrape/register.cy.js [hunk 11]
```diff
@@ -1,162 +1,167 @@
...
         cy.get(':nth-child(5) > :nth-child(3) > .table-plan-text').contains('$349.99 / Monthly');
     });
 
-    it('11. Test to Verify the "Features" button of the Free plan and should open the popup for free plan', () => {
+    it('11. Test to verify the "Features" button of the "Free" monthly plan and should open the popup for "Free" monthly plan', () => {
         cy.get(':nth-child(1) > :nth-child(2) > .btn').contains('Features').click({ force: true });
         cy.get('.modal-header').contains('Free');
         cy.get('.plan-feature-list > li').should('be.visible', 'contain.text', '1.000 requests per month');
         cy.get('.modal-footer > .btn').click({ force: true });
         cy.wait(500);
```
Comment (NaveenK15) [inline cypress/e2e/zenscrape/register.cy.js]: explain??

File: cypress/e2e/zenscrape/register.cy.js [hunk 12]
```diff
@@ -1,162 +1,167 @@
...
     });
 
-    it('18. Test to Verify the "Features" button of the Free plan and should open the popup for free plan', () => {
+    it('18. Test to verify the "Features" button of the "Free" plan and should open the popup for "Free" yearly plan', () => {
         cy.get(':nth-child(1) > :nth-child(2) > .btn').contains('Features').click({ force: true });
-        cy.get('.modal-header').contains('Free');
+        cy.get('.modal-header').contains('Free').should('be.visible');
         cy.get('.plan-feature-list > li').should('be.visible', 'contain.text', '1.000 requests per month');
         cy.get('.modal-footer > .btn').click({ force: true });
         cy.wait(500);
```
Comment (NaveenK15) [inline cypress/e2e/zenscrape/register.cy.js]: explain?

File: cypress/e2e/zenscrape/register.cy.js [hunk 13]
```diff
@@ -1,162 +1,167 @@
...
     });
 
-    it('19. Test to Verify the "Features" button of Small Yearly plan and should open the popup for Small Yearly plan', () => {
+    it('19. Test to verify the "Features" button of "Small" yearly plan and should open the popup for "Small" yearly plan', () => {
         cy.get(':nth-child(2) > :nth-child(2) > .btn').contains('Features').click({ force: true })
-        cy.get('.modal-header').contains('Small Yearly');
+        cy.get('.modal-header').contains('Small Yearly').should('be.visible');
         cy.get('.plan-feature-list > li').should('be.visible', 'contain.text', '250 000 requests per month');
         cy.get('.modal-footer > .btn').click({ force: true });
         cy.wait(500);
```
Comment (NaveenK15) [inline cypress/e2e/zenscrape/register.cy.js]: explain

File: cypress/e2e/zenscrape/register.cy.js [hunk 14]
```diff
@@ -1,162 +1,167 @@
...
     });
 
-    it('20. Test to Verify the "Features" button of Medium Yearly plan and should open the popup for Medium Yearly plan', () => {
+    it('20. Test to verify the "Features" button of "Medium" yearly plan and should open the popup for "Medium" yearly plan', () => {
         cy.get(':nth-child(3) > :nth-child(2) > .btn').contains('Features').click({ force: true });
-        cy.get('.modal-header').contains('Medium');
+        cy.get('.modal-header').contains('Medium').should('be.visible');
         cy.get('.plan-feature-list > li').should('be.visible', 'contain.text', '1.000.000 requests per month');
         cy.get('.modal-footer > .btn').click({ force: true });
         cy.wait(500);
```
Comment (NaveenK15) [inline cypress/e2e/zenscrape/register.cy.js]: explain

File: cypress/e2e/zenscrape/register.cy.js [hunk 15]
```diff
@@ -1,162 +1,167 @@
...
+        //Todo: Added an 0.5 second wait to load the features pop up. A more reliable solution should be implemented to handle this wait.
     });
 
-    it('22. Test to Verify the "Features" button of Professional Yearly plan and should open the popup for Professional Yearly plan', () => {
+    it('22. Test to verify the "Features" button of "Professional" yearly plan and should open the popup for "Professional" yearly plan', () => {
         cy.get(':nth-child(5) > :nth-child(2) > .btn').contains('Features').click({ force: true });
-        cy.get('.modal-header').contains('Professional');
+        cy.get('.modal-header').contains('Professional').should('be.visible');
         cy.get('.plan-feature-list > li').should('be.visible', 'contain.text', '5 000 000 requests per month');
         cy.wait(500);
```
Comment (NaveenK15) [inline cypress/e2e/zenscrape/register.cy.js]: explain

File: cypress/e2e/zenscrape/register.cy.js [hunk 16]
```diff
@@ -1,162 +1,167 @@
...
     });
 
-    it('21. Test to Verify the "Features" button of Large Yearly plan and should open the popup for Large Yearly plan', () => {
+    it('21. Test to verify the "Features" button of "Large" yearly plan and should open the popup for "Large" yearly plan', () => {
         cy.get(':nth-child(4) > :nth-child(2) > .btn').contains('Features').click({ force: true });
-        cy.get('.modal-header').contains('Large');
+        cy.get('.modal-header').contains('Large').should('be.visible');
         cy.get('.plan-feature-list > li').should('be.visible', 'contain.text', '3.000.000 requests per month');
         cy.get('.modal-footer > .btn').click({ force: true });
         cy.wait(500);
```
Comment (NaveenK15) [inline cypress/e2e/zenscrape/register.cy.js]: ?

File: cypress/e2e/zenscrape/register.cy.js (hunk 15 above)
Comment (NaveenK15) [inline cypress/e2e/zenscrape/register.cy.js]: ?

File: cypress/e2e/zenscrape/status.cy.js [hunk 17]
```diff
@@ -0,0 +1,138 @@
...
+    it('8. Test to check the "Subscribe" button present', () => {
+        cy.componentVisiblityCheck('.subscribe');
+    });
+
+    it('9. Test to check by clicing the "Subscribe" button', () => {
+        cy.get('.subscribe').click();
+    });
+
+    it('10. Test to check the "Email" text, "Get notified when zenscrape creates, updates or resolves an incident" caption, "Email" input field and the "Subscribe" button present', () => {
+        cy.wait(4000);
```
Comment (NaveenK15) [inline cypress/e2e/zenscrape/status.cy.js]: ?

**Review by NaveenK15 — APPROVED** [review]

**Pull request merged**